app/domain/authentication/authn_gcp/validate_resource_restrictions_match_jwt.rb

```diff
@@ -0,0 +1,60 @@
+module Authentication
+  module AuthnGcp
+
+    # This class is responsible of validating resource restrictions values against JWT token.
+    # the assumption is that resource restrictions list contains only permitted types which already validated
+    # in previous steps
+    ValidateResourceRestrictionsMatchJWT = CommandClass.new(
+      dependencies: {
+        logger: Rails.logger
+      },
+      inputs:       %i(resource_restrictions decoded_token restriction_prefix)
+    ) do
+
+      def call
+        @logger.debug(LogMessages::Authentication::AuthnGcp::ValidatingResourceRestrictionsValues.new)
+        validate
+        @logger.debug(LogMessages::Authentication::AuthnGcp::ValidatedResourceRestrictionsValues.new)
+      end
+
+      private
+
+      def validate
+        @resource_restrictions.each do |r|
+          resource_type = r.type
+          resource_value = r.value
+
+          case remove_prefix(resource_type)
+          when PROJECT_ID_RESTRICTION_NAME
+            unless resource_value == @decoded_token.project_id
+              raise Errors::Authentication::AuthnGcp::InvalidResourceRestrictions.new(resource_type)
+            end
+            @logger.debug(LogMessages::Authentication::ValidatedResourceRestrictionValue.new(resource_type))
+
+          when INSTANCE_NAME_RESTRICTION_NAME
+            unless resource_value == @decoded_token.instance_name
+              raise Errors::Authentication::AuthnGcp::InvalidResourceRestrictions.new(resource_type)
+            end
+            @logger.debug(LogMessages::Authentication::ValidatedResourceRestrictionValue.new(resource_type))
+
+          when SERVICE_ACCOUNT_ID_RESTRICTION_NAME
+            unless resource_value == @decoded_token.service_account_id
+              raise Errors::Authentication::AuthnGcp::InvalidResourceRestrictions.new(resource_type)
+            end
+            @logger.debug(LogMessages::Authentication::ValidatedResourceRestrictionValue.new(resource_type))
+
+          when SERVICE_ACCOUNT_EMAIL_RESTRICTION_NAME
+            unless resource_value == @decoded_token.service_account_email
+              raise Errors::Authentication::AuthnGcp::InvalidResourceRestrictions.new(resource_type)
+            end
+            @logger.debug(LogMessages::Authentication::ValidatedResourceRestrictionValue.new(resource_type))
+          end
+        end
+      end
+
+      def remove_prefix(restriction_type)
```

Comment:
when methods return an object we give it the name of the returned object. so i would change this to `restriction_type_without_prefix` or `non-prefixed-restriction_type`.